File: src/main/java/de/danielmaile/aether/item/ListenerCrafting.java
package de.danielmaile.aether.item;

import de.danielmaile.aether.util.NBTEditor;
import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.PrepareItemCraftEvent;
import org.bukkit.inventory.ItemStack;

public class ListenerCrafting implements Listener
{
    @EventHandler
    //Check for crafting of vanilla items with aether items
    public void onCraftPrepare(PrepareItemCraftEvent event)
    {
        ItemStack result = event.getInventory().getResult();
        if (result == null) return;

        if (NBTEditor.hasKey(result, CustomItemType.AETHER_ITEM_TAG_KEY)) return;

        ItemStack[] ingredients = event.getInventory().getContents();

        for (ItemStack ingredient : ingredients)
        {
            if (ingredient == null || ingredient.getType() == Material.AIR) continue;

            if (NBTEditor.hasKey(ingredient, CustomItemType.AETHER_ITEM_TAG_KEY))
            {
                event.getInventory().setResult(null);
                return;
            }
        }
    }
}

File: src/main/java/de/danielmaile/aether/item/ListenerCrafting.kt
package de.danielmaile.aether.item

import de.danielmaile.aether.util.hasKey
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.PrepareItemCraftEvent

class ListenerCrafting : Listener {

    //Check for crafting of vanilla items with aether items
    @EventHandler
    fun onCraftPrepare(event: PrepareItemCraftEvent) {
        val result = event.inventory.result ?: return
        if (result.hasKey(AETHER_ITEM_TAG_KEY)) return
        val ingredients = event.inventory.contents ?: return

        for (ingredient in ingredients) {
            if (ingredient == null || ingredient.type == Material.AIR) continue
            if (ingredient.hasKey(AETHER_ITEM_TAG_KEY)) {
                event.inventory.result = null
                return
            }
        }
    }
}